src/software/ai/hl/stp/play/penalty_kick_play_test.cpp

```diff
@@ -12,32 +17,80 @@ class PenaltyKickPlayTest : public SimulatedPlayTestFixture
 {
 };
 
-TEST_F(PenaltyKickPlayTest, test_penalty_kick_play)
+TEST_F(PenaltyKickPlayTest, test_penalty_kick_setup)
 {
-    setBallState(BallState(field().penaltyFriendly(), Vector(0, 0)));
+    setBallState(BallState(field().friendlyPenaltyMark(), Vector(0, 0)));
     addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId(
-        {Point(-3, 2.5), Point(-3, 1.5), Point(-3, 0.5), Point(-3, -0.5), Point(-3, -1.5),
-         Point(4.6, -3.1)}));
+        {Point(-2, -2), Point(-3, -1), Point(-3, 0), Point(-3, 1), Point(-3, 2),
+         Point(2, 2.5)}));
     setFriendlyGoalie(0);
     addEnemyRobots(
         TestUtil::createStationaryRobotStatesWithId({field().enemyGoalCenter()}));
     setEnemyGoalie(0);
     setAIPlay(TYPENAME(PenaltyKickPlay));
-    setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::PREPARE_PENALTY_US);
+    setRefereeCommand(RefereeCommand::PREPARE_PENALTY_US, RefereeCommand::NORMAL_START);
 
+    RobotId shooter_id                                               = 5;
     std::vector<ValidationFunction> terminating_validation_functions = {
-        // This will keep the test running for 9.5 seconds to give everything enough
-        // time to settle into position and be observed with the Visualizer
-        // TODO: Implement proper validation
-        // https://github.com/UBC-Thunderbots/Software/issues/1396
-        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+        [shooter_id](std::shared_ptr<World> world_ptr,
+                     ValidationCoroutine::push_type& yield) {
+            // Wait 2 seconds for robots to start moving adequately far away from the ball
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(2))
             {
                 yield();
             }
+            robotsAvoidBall(1, {shooter_id}, world_ptr, yield);
+        },
```

Comment:
please move this to the `non_terminating_validation_functions` vector

Reply:
done